__init__.py

```diff
@@ -51,6 +51,51 @@ class PairingMode(str, Enum):
     GUI = "gui"  # gui - click buttons
 
 
+
+class GUIHelper:
+    """ helper class for better handling of specific setup GUI like list"""
+
+    def __init__(self) -> None:
+        pass
+
+    def distinguish_models_tts(self, model):
+        """ distinguish the model to a separate list for engines and voices
+        to build a nested list for the GUI """
+        engines_model = []
+
+        for entry in model:
+            if entry["engine"] not in [engine["engine"] for engine in engines_model]:
+                engines_model.append({"engine": entry["engine"], "plugin_name": entry["plugin_name"], "supports_male_voice": False, "supports_female_voice": False, "supports_offline_mode": False, "supports_online_mode": False,  "voiceModel": []})
```

Comment:
this looks wrong, we have a new UI api coming directly from opm, this should be added there instead

please give me a data model with example settings you need, here you are adding a bunch of random stuff such as "supports_XXX" that should be coming directly from OPM instead

Reply:
This is not random stuff, this is directly coming from the api in the setup skill

It's only evaluating:

{"plugin_name": plugin_display_name, "display_name": voice.get("display_name", " "), "gender": voice.get("gender", " "), "offline": voice.get("offline", False), "lang": lang, 'engine': engine}

To build a new list that will be displayed as the engine models list that simply states after evaluating the voices in the engine whether the engine supports male or female or none gender voices and whether the engine supports both online and offline voices

This does not add anything new, this is a QML GUI listview model